tickets/src/main/java/no/unit/nva/publication/ticket/create/TicketResolver.java

```diff
@@ -0,0 +1,145 @@
+import static no.unit.nva.publication.ticket.create.CreateTicketHandler.BACKEND_CLIENT_SECRET_NAME;
+import static nva.commons.core.attempt.Try.attempt;
+
+import java.net.URI;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import no.unit.nva.model.Publication;
+import no.unit.nva.model.associatedartifacts.AssociatedArtifact;
+import no.unit.nva.model.associatedartifacts.AssociatedArtifactList;
+import no.unit.nva.model.associatedartifacts.file.File;
+import no.unit.nva.publication.exception.TransactionFailedException;
+import no.unit.nva.publication.external.services.AuthorizedBackendUriRetriever;
+import no.unit.nva.publication.external.services.RawContentRetriever;
+import no.unit.nva.publication.model.business.PublishingRequestCase;
+import no.unit.nva.publication.model.business.PublishingWorkflow;
+import no.unit.nva.publication.model.business.TicketEntry;
+import no.unit.nva.publication.model.business.TicketStatus;
+import no.unit.nva.publication.model.business.UserInstance;
+import no.unit.nva.publication.service.impl.ResourceService;
+import no.unit.nva.publication.service.impl.TicketService;
+import no.unit.nva.publication.ticket.model.identityservice.CustomerTransactionResult;
+import nva.commons.apigateway.exceptions.ApiGatewayException;
+import nva.commons.apigateway.exceptions.BadGatewayException;
+import nva.commons.core.JacocoGenerated;
+
+public class TicketResolver {
+
+    public static final String CONTENT_TYPE = "application/json";
+    private final ResourceService resourceService;
+    private final TicketService ticketService;
+    private final RawContentRetriever uriRetriever;
+
+    @JacocoGenerated
+    public TicketResolver() {
+        this(ResourceService.defaultService(), TicketService.defaultService(),
+                new AuthorizedBackendUriRetriever(BACKEND_CLIENT_SECRET_NAME, BACKEND_CLIENT_AUTH_URL));
+    }
+
+    public TicketResolver(ResourceService resourceService, TicketService ticketService,
+                          RawContentRetriever uriRetriever) {
+        this.resourceService = resourceService;
+        this.ticketService = ticketService;
+        this.uriRetriever = uriRetriever;
+    }
+
+    public TicketEntry resolveAndPersistTicket(TicketEntry ticket, Publication publication, URI customerId) throws ApiGatewayException {
+        if (isPublishingRequestCase(ticket)) {
+            var customerTransactionResult = getCustomerTransactionResult(customerId);
+
+            if (customerTransactionResult.customerAllowsRegistratorsToPublishDataAndMetadata()) {
+                ((PublishingRequestCase) ticket).setWorkflow(PublishingWorkflow.REGISTRATOR_PUBLISHES_METADATA_AND_FILES);
+                var persistedTicket = persistTicket(ticket);
+                approveTicketAndPublishPublication(persistedTicket, publication);
+                return persistedTicket;
+            }
+        }
+        return persistTicket(ticket);
```

Comment:
Bra, jeg hadde kanskje forkortet denne metoden noe, for eksempel:

    public TicketEntry resolveAndPersistTicket(TicketEntry ticket, Publication publication, URI customerId) throws ApiGatewayException {
        if (isPublishingRequestCase(ticket)) {
            return resolvePublishingRequest(ticket, publication, customerId);
        }
        return persistTicket(ticket);
    }